src/packet.rs

```diff
@@ -89,7 +91,7 @@ macro_rules! packet {
                 $name { $( $fieldname: read_packet_field!(stream, $datatype) ),* }
             }
             pub fn write<S: MinecraftProtocolWriter>(&self, stream: &mut S) {
-                $( write_packet_field!(stream, self.$fieldname, $datatype) );*
+                $( write_packet_field!(stream, self.$fieldname.clone(), $datatype) );*

```

Comment:
Now that we have fields that don't live purely on the stack (the vectors) we can't just copy the arguments in anymore and have to clone them instead. This is obviously a performance hit but I'm prioritizing functionality over performance here in the context of this course. We could come back later and make sure that all the types are being written efficiently if we really wanted to